homeassistant/components/hunterdouglas_powerview/sensor.py

```diff
@@ -30,43 +32,65 @@ async def async_setup_entry(
 
     pv_entry: PowerviewEntryData = hass.data[DOMAIN][entry.entry_id]
 
-    entities = []
+    entities: list[PowerViewSensor] = []
     for raw_shade in pv_entry.shade_data.values():
         shade: BaseShade = PvShade(raw_shade, pv_entry.api)
-        if SHADE_BATTERY_LEVEL not in shade.raw_data:
-            continue
         name_before_refresh = shade.name
         room_id = shade.raw_data.get(ROOM_ID_IN_SHADE)
         room_name = pv_entry.room_data.get(room_id, {}).get(ROOM_NAME_UNICODE, "")
-        entities.append(
-            PowerViewShadeBatterySensor(
-                pv_entry.coordinator,
-                pv_entry.device_info,
-                room_name,
-                shade,
-                name_before_refresh,
+        if SHADE_BATTERY_LEVEL in shade.raw_data:
```

Comment:
The key and class could be a dict mapping here since arguments to the constructors are the same.  

You may need to move async_setup_entry to the end of the file for the class names to be storable in a dict

Reply:
done